igor-web/src/main/groovy/com/netflix/spinnaker/igor/gitlabci/service/GitlabCiService.java

```diff
@@ -77,13 +77,13 @@ public List<GenericGitRevision> getGenericGitRevisions(String job, GenericBuild
   }
 
   @Override
-  public GenericBuild getGenericBuild(String projectId, int pipelineId) {
+  public GenericBuild getGenericBuild(String projectId, String pipelineId) {
     Project project = client.getProject(projectId);
     if (project == null) {
       log.error("Could not find Gitlab CI Project with projectId={}", projectId);
       return null;
     }
-    Pipeline pipeline = client.getPipeline(projectId, pipelineId);
+    Pipeline pipeline = client.getPipeline(projectId, Long.valueOf(pipelineId));
```

Comment:
Won't there be a struggle if pipelineId isn't numeric here?

Reply:
agreed; assumption is that gitlab only has numeric pipelineId. we could add a precondition assertion to that effect, perhaps?

Reply:
Long.valueOf throws an exception...but I bet a custom check/log message would be helpful.  Are there some gitlab docs that talk about this though?  If it turns out pipelineIds can be non-numeric, this'll be a behavior change, won't it?